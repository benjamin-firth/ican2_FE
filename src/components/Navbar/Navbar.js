import React from 'react';
import './Navbar.scss';
import NavButton from '../NavButton/NavButton';
import simpleLogo from '../../images/simple-logo.png';

const Navbar = ({ buttons }) => {

  const renderButtons = () => {
    return buttons.map(button => (
      <NavButton nav={button.nav} name={button.name} />
    ))
  }

  return (
    <nav>
      <img src={simpleLogo} />
      <div className='nav-button-container'>
        {renderButtons()}
      </div>
    </nav>
  );
}

export default Navbar;
