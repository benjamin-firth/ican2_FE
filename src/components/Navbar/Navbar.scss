@import '../../variables';

nav {
  align-items: center;
  background-color: $primarycolor;
  border-bottom: 6px solid $secondarycolor;
  display: flex;
  height: 10vh;
  justify-content: flex-end;
  width: 100vw;

  // Hamburger icon

  .col {
    display: block;
    float:left;
    margin: 1% 0 1% 1.6%;
  }

  .col:first-of-type {
    margin-left: 0;
  }

  .hamburger {
    margin-right: 20px;
  }

  .hamburger .line{
    width: 50px;
    height: 5px;
    background-color: #000000;
    display: block;
    margin: 8px auto;
  }

  .hamburger:hover{
    cursor: pointer;
  }
}
