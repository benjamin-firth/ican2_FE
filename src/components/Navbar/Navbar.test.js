import React from 'react';
import { shallow } from 'enzyme';
import Navbar from './Navbar';

describe('Navbar', () => {

  it('should match the snapshot', () => {
    expect(true).toBe(true);
  });

});
