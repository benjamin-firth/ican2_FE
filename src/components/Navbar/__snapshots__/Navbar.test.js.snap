// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should match the snapshot 1`] = `
<nav
  className="main-nav"
>
  <img
    src="simple-logo.png"
  />
  <div
    className="nav-button-container"
  >
    <NavButton
      name="find mentors"
      nav="/mentors"
    />
    <NavButton
      name="messages"
      nav="/messages"
    />
    <NavButton
      name="meetups"
      nav="/meetups"
    />
    <NavButton
      name="my profile"
      nav="/myprofile"
    />
    <NavButton
      name="log out"
      nav="/"
    />
  </div>
</nav>
`;
